src/CMakeLists.txt

```diff
@@ -161,51 +159,71 @@ if(ENABLE_POSTGRESQL)
 			set(PostgreSQL_INCLUDE_DIRS ${PostgreSQL_INCLUDE_DIR})
 		endif()
 	else()
-		find_package(PostgreSQL)
+		find_package(PostgreSQL QUIET)
 	endif()
+endmacro()
+
+
+set(ENABLE_POSTGRESQL SHRUG CACHE STRING "Enable PostgreSQL backend")
+set_property(CACHE ENABLE_POSTGRESQL PROPERTY STRINGS SHRUG ON OFF)
```

Comment:
my suggestion to keep compatibility with how these options worked before:
* REQUIRE = search and error if missing
* ON / TRUE / `1` = search but dont error
* OFF / FALSE / `0` = dont search

Reply:
One of the objectives of the PR is to change the way this works. Description has been updated.

Reply:
I see there's a potentially big problem here because it could break compatibility with people's build scripts. It would probably be almost entirely alleviated just by allowing `REQUIRED` and making it an alias for `ON`, but perhaps it's still a bit unprecedented. What are your thoughts on this? Is it worth making the build system simpler if the options no longer work the same way? Maybe it would be possible to allow setting an option to switch to the new behavior, and then deprecate the old behavior after 1 or 2 more major releases or something.